internal/pkg/cli/app_show.go

```diff
@@ -72,109 +62,100 @@ type ShowAppOpts struct {
 }
 
 // Ask asks for fields that are required but not passed in.
-func (opts *ShowAppOpts) Ask() error {
-	// get project
-	if opts.ProjectName() == "" {
-		projectName, err := opts.selectProject()
+func (o *ShowAppOpts) Ask() error {
+	if err := o.askProject(); err != nil {
+		return err
+	}
+	return o.askAppName()
+}
+
+// Validate returns an error if the values provided by the user are invalid.
+func (o *ShowAppOpts) Validate() error {
+	if o.ProjectName() != "" {
+		names, err := o.retrieveProjects()
```

Comment:
We already called `o.retrieveProjects` in the `askProject` method. Can we stash the result from the first call instead of making another network call?